third_party/terraform/utils/deployment_manager_operation.go

```diff
@@ -0,0 +1,84 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"google.golang.org/api/compute/v1"
+)
+
+type DeploymentManagerOperationWaiter struct {
```

Comment:
Since this is a somewhat normal looking operation waiter (aside from errors), can this be auto-genned?

Reply:
Eventually, I'd like to! The big issue is that we'd have to create a different autogen operation template that uses the Compute Operation instead of CommonOperation (cloudresourcemanager.Operation = CommonOperation).  Two key differences:

- CommonOp status is done = true/false, compute.Op is status DONE/PENDING/RUNNING
- common op error is op.Error, compute has HTTPErrorMessage, HTTPErrorCode, and Error.Errors 

so the options for autogen are:
1) Create a new autogen operation template using compute.Operation, which I'd do in a seperate PR anyways and hopefully reuse in ComputeOperationWaiter
2) Change the current resource template to just use the compute methods instead of `<%= resourceName%>OperationWait`
3) Add a handwritten operation for DeploymentManagerOperationWaiter that inherits ComputeOperationWaiter <<<<< What I did

The reason for now I want to do #3 is that I'm pretty sure we will want to have custom DeploymentManager-specific warnings and error messages, especially if someone runs into a confusing but expected behavior.